src/lib/structures/Event.ts

```diff
@@ -22,7 +22,7 @@ export abstract class Event extends BasePiece {
 		this.#listener = this.emitter && this.event ? (this.once ? this._runOnce.bind(this) : this._run.bind(this)) : null;
 	}
 
-	public abstract run(...args: readonly any[]): unknown;
+	public abstract run(...args: E extends keyof ClientEvents ? ClientEvents[E] : any[]): unknown;
```

Comment:
Don't we technically want to type this as "unknown[]" instead? ditto with the ClientEvents interface

Reply:
This is more of a @kyranet question as it was his choice to use any over unknow.